packages/mui-material/src/MobileStepper/MobileStepper.js

```diff
@@ -76,31 +103,38 @@ const MobileStepperDot = styled('div', {
 
     return [styles.dot, dotActive && styles.dotActive];
   },
-})(({ theme, ownerState, dotActive }) => ({
-  ...(ownerState.variant === 'dots' && {
-    transition: theme.transitions.create('background-color', {
-      duration: theme.transitions.duration.shortest,
-    }),
-    backgroundColor: (theme.vars || theme).palette.action.disabled,
-    borderRadius: '50%',
-    width: 8,
-    height: 8,
-    margin: '0 2px',
-    ...(dotActive && {
-      backgroundColor: (theme.vars || theme).palette.primary.main,
-    }),
-  }),
+})(({ theme }) => ({
+  variants: [
+    {
+      props: { variant: 'dots', dotActive: false },
+      style: {
+        ...mobileStepperDotDefaultStyle({ theme }),
+      },
+    },
+    {
+      props: { variant: 'dots', dotActive: true },
+      style: {
+        ...mobileStepperDotDefaultStyle({ theme }),
+        backgroundColor: (theme.vars || theme).palette.primary.main,
+      },
+    },
+  ],
```

Comment:
No need to abstract the styles.

```suggestion
  variants: [
    {
      props: { variant: 'dots' },
      style: {
        transition: theme.transitions.create('background-color', {
        duration: theme.transitions.duration.shortest,
      }),
      backgroundColor: (theme.vars || theme).palette.action.disabled,
      borderRadius: '50%',
      width: 8,
      height: 8,
      margin: '0 2px',
      },
    },
    {
      props: { variant: 'dots', dotActive: true },
      style: {
        backgroundColor: (theme.vars || theme).palette.primary.main,
      },
    },
  ],
```